lib/services/elasticsearch.js

```diff
@@ -153,665 +173,682 @@ class ElasticSearch extends Service {
+   * @param {string} collection - Collection name
+   *
+   * @returns {Promise} { }
    */
-  truncateCollection(request) {
-    const
-      deleteRequest = new Request({
-        index: request.input.resource.index,
-        collection: request.input.resource.collection,
-        body: {
-          query: {
-            match_all: {}
-          }
-        }
-      }, {
-        user: request.context.user
-      });
+  truncateCollection (index, collection) {
+    let mappings;
 
-    deleteRequest.input.args.refresh = request.input.args.refresh || false;
+    const esRequest = {
+      index: this._getESIndex(index, collection)
+    };
+
+    return this.getMapping(index, collection)
+      .then(collectionMappings => {
+        mappings = collectionMappings;
+
+        return this.client.indices.delete(esRequest);
+      })
+      .then(() => this.client.indices.create({ ...esRequest, mappings }))
+      .then(() => {
+        const apiResponse = {};
 
-    return this.deleteByQuery(deleteRequest)
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+        return apiResponse;
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Run several action and document
    *
-   * @param {Request} request
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {Array<object>} documents - Documents to import
+   * @param {object} options - timeout (null), refresh (false)
+   *
    * @returns {Promise}
    */
-  import(request) {
+  import (
+    index,
+    collection,
+    documents,
+    { refresh=null, timeout=null, userId=null } = {})
+  {
     const
+      esRequest = {
+        index: this._getESIndex(index, collection),
+        body: documents,
+        refresh,
+        timeout
+      },
       actionNames = ['index', 'create', 'update', 'delete'],
-      esRequest = initESRequest(request,
-        ['consistency', 'refresh', 'timeout', 'fields']),
-      userId = this._getUserId(request),
-      dateNow = Date.now();
+      dateNow = Date.now(),
+      kuzzleMeta = {
+        created: {
+          author: String(userId),
+          createdAt: dateNow,
+          updatedAt: null,
+          updater: null
+        },
+        updated: {
+          updater: String(userId),
+          updatedAt: dateNow
+        }
+      };
 
     assertWellFormedRefresh(esRequest);
 
-    if (!request.input.body || !(request.input.body.bulkData instanceof Object)) {
-      return Bluebird.reject(
-        this.getError('missing_or_invalid_import_attribute')
-      );
-    }
-
-    esRequest.body = request.input.body.bulkData;
-    const kuzzleMetaCreated = {
-      author: userId,
-      createdAt: dateNow,
-      updatedAt: null,
-      updater: null,
-      active: true,
-      deletedAt: null
-    };
-    const kuzzleMetaUpdated = {
-      updater: userId,
-      updatedAt: dateNow
-    };
-
+    // TODO use indexCache.indexes instead
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)